packages/renderer/src/AppNavigation.spec.ts

```diff
@@ -0,0 +1,46 @@
+import '@testing-library/jest-dom';
+import { beforeAll, test, expect, vi } from 'vitest';
+import { render, screen } from '@testing-library/svelte';
+import AppNavigation from './AppNavigation.svelte';
+
+// fake the window.events object
+beforeAll(() => {
+  (window.events as unknown) = {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    receive: (_channel: string, func: any) => {
+      func();
+    },
+  };
+});
+
+async function waitRender(customProperties: object): Promise<void> {
+  const result = render(AppNavigation, { ...customProperties });
+  // wait that result.component.$$.ctx[2] is set
+  while (result.component.$$.ctx[2] === undefined) {
+    await new Promise(resolve => setTimeout(resolve, 100));
+  }
+}
+
```

Comment:
do we need the internal waitRender or we can use the original render as I'm not sure here we need to wait the end of onMount method